### PR TITLE
tikv: add quota limiter doc in config

### PR DESCRIPTION
**Pull request by hunterlxt**
Signed-off-by: Xintao <hunterlxt@live.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add TiKV quota related config doc.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/tikv/rfcs/pull/89 https://github.com/tikv/tikv/pull/12151

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by hunterlxt** [comment]
@en-jin19 PTAL

**Comment by en-jin19** [comment]
@hunterlxt Could you please invite a tech reviewer? thank you!

**Comment by hunterlxt** [comment]
PTAL @tonyxuqqi

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
+
+### `forefront-cpu-time`
+
++ 限制处理前端请求的 CPU 耗时，单位为 millicpu，即 1500 代表 1.5vCPU。
++ 默认值：0，即不设限
+
+### `forefront-req-rate`
+
++ 限制处理前端请求的速率，被限制的前端请求仅包含事务读写相关的请求。
++ 默认值：0，即不设限
```
Comment (en-jin19) [inline tikv-configuration-file.md:1589]:
```suggestion
### `forefront-cpu-time`

+ 限制 CPU 为处理前端请求所花费的时间。
+ 默认值：0（即无限制）
+ 单位：millicpu（即当该参数值为 `1500` 时，前端请求会消耗 1.5vCPU）

### `forefront-req-rate`

+ 限制 TiKV 处理前端的请求（**仅限**事务的读写请求）的速率。
+ 默认值：0（即无限制）
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
+
+### `write-bandwidth`
+
++ 限制写入带宽，被限制的写入带宽仅包含事务写入的部分。
++ 默认值：0kb，即不设限
+
+### `read-bandwidth`
+
++ 限制读取带宽，被限制的读取带宽仅包含事务读和 coprocessor 相关的部分。
++ 默认值：0kb，即不设限
```
Comment (en-jin19) [inline tikv-configuration-file.md:1604]:
```suggestion
### `write-kvs` <span class="version-mark">从 v6.0 版本开始引入</span>

+ 限制事务写入键值对的速率。
+ 默认值：0（即无限制）

### `write-bandwidth` <span class="version-mark">从 v6.0 版本开始引入</span>

+ 限制事务写入的带宽。
+ 默认值：0KB（即无限制）

### `read-bandwidth` <span class="version-mark">从 v6.0 版本开始引入</span>

+ 限制事务读取数据和 Coprocessor 读取数据的带宽。
+ 默认值：0KB（即无限制）
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
 + 开启流水线式加悲观锁流程。开启该功能后，TiKV 在检测数据满足加锁要求后，立刻通知 TiDB 执行后面的请求，并异步写入悲观锁，从而降低大部分延迟，显著提升悲观事务的性能。但有较低概率出现悲观锁异步写入失败的情况，可能会导致悲观事务提交失败。
 + 默认值：true
+
+## quota
+
+前端限流的相关配置，以下参数的限制采用近似统计。
+
+> **警告：**
+>
+> Quota Limiter 目前为实验特性，设置较小的限额可能发生性能下降的现象。
```
Comment (en-jin19) [inline tikv-configuration-file.md:1579]:
```suggestion
用于前台限流相关的配置项。

当【XXX，XXX 等物理资源有限】  时，如果 TiKV 前台处理的读写请求量过大，会导致前台和后台处理请求时占用彼此的 CPU 资源，最终影响 TiKV 性能的稳定性。此时，你可以通过开启前台限流 (Quota Limiter) 相关的 quota 配置项限制前台各类请求所占用的资源。

> **警告：**
>
> - 前台限流是 TiDB 在 v6.0 中引入的实验特性，不建议在生产环境中使用。
> - 如果 quota 相关配置项的值较小（【XXX举例给出一个具体数字，提示用户多小是“较小”】），可能发生 TiKV 性能下降的情况。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
+
+### `forefront-cpu-time`
+
++ 限制处理前端请求的 CPU 耗时，单位为 millicpu，即 1500 代表 1.5vCPU。
```
Comment (en-jin19) [inline tikv-configuration-file.md:1583]: 这里的单位 “millicpu” ，是不是“CPU 资源使用量”的单位，而不是"时间（耗时）“的单位呢？是否需要修改描述内容或单位？

Comment (hunterlxt) [reply tikv-configuration-file.md:1583]: 好的

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1569,3 +1569,31 @@ Raft Engine 相关的配置项。
 
 + 开启流水线式加悲观锁流程。开启该功能后，TiKV 在检测数据满足加锁要求后，立刻通知 TiDB 执行后面的请求，并异步写入悲观锁，从而降低大部分延迟，显著提升悲观事务的性能。但有较低概率出现悲观锁异步写入失败的情况，可能会导致悲观事务提交失败。
 + 默认值：true
+
+## quota
+
+用于前台限流相关的配置项。
```
Comment (tonyxuqqi) [inline tikv-configuration-file.md:1575]: 加上实验功能的说明

**Comment by tonyxuqqi** [comment]
add experimental feature statement in the doc.

**Comment by hunterlxt** [comment]
> add experimental feature statement in the doc.

@tonyxuqqi L1581 这个可以吗

**Comment by tonyxuqqi** [comment]
LGTM

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
+
+当 TiKV 部署的机型资源受限时，如果前台处理的读写请求量过大，会导致后台处理请求的 CPU 资源被前台占用，最终影响 TiKV 性能的稳定性。此时，你可以通过开启前台限流 (Quota Limiter) 相关的 quota 配置项限制前台各类请求所占用的资源。
+
+> **警告：**
+>
+> - 前台限流是 TiDB 在 v6.0 中引入的实验特性，不建议在生产环境中使用。
+> - 该功能适合在资源有限的环境中（如 4vCPU，16G 内存）提高长期运行的稳定性，但启用该功能则会在资源丰富的机型环境出现 TiKV 峰值性能下降的情况。
+
+### `foreground-cpu-time` <span class="version-mark">从 v6.0 版本开始引入</span>
```
Comment (TomShawn) [inline tikv-configuration-file.md:1584]: @en-jin19 前端不支持 `<span class="version-mark">` 的锚点链接
```suggestion
### `foreground-cpu-time`（从 v6.0 版本开始引入）
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1569,3 +1569,36 @@ Raft Engine 相关的配置项。
...
+
++ 限制 CPU 为处理 TiKV 前台读写请求所使用的 CPU 资源使用量，这是一个软限制。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1586]:
```suggestion
+ 限制处理 TiKV 前台读写请求所使用的 CPU 资源使用量，这是一个软限制。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by en-jin19 — APPROVED** [review]
LGTM

**Comment by en-jin19** [comment]
> LGTM

Hi @tonyxuqqi, could you please approve this PR? thank you!

**Review by tonyxuqqi — APPROVED** [review]

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -1574,3 +1574,35 @@ Raft Engine 相关的配置项。
...
+## quota
+
+用于前台限流 (Quota Limiter) 相关的配置项。
+
+当 TiKV 部署的机型资源有限（如 4v CPU，16 G 内存）时，如果 TiKV 前台处理的读写请求量过大，会占用 TiKV 后台处理请求所需的 CPU 资源，最终影响 TiKV 性能的稳定性。此时，你可以使用前台限流相关的 quota 配置项以限制前台各类请求占用的 CPU 资源。触发该限制的请求会被强制等待一段时间以让出 CPU 资源。具体等待时间与新增请求量相关，最多不超过 [`max-delay-duration`](#max-delay-duration从-v600-版本开始引入) 的值。
+
+> **警告：**
+>
+> - 前台限流是 TiDB 在 v6.0.0 中引入的实验特性，不建议在生产环境中使用。
+> - 该功能仅适合在资源有限的环境中使用，以保证 TiKV 在该环境下可以长期稳定地运行。如果在资源丰富的机型环境中开启该功能，可能会导致读写请求量达到峰值时 TiKV 的性能下降的问题。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1587]: 感觉”长期“比较模糊。多长是”长期“呢？

**Pull request merged**